common/taskana-common/src/main/java/pro/taskana/common/internal/util/ObjectAttributeChangeDetector.java

```diff
@@ -79,26 +52,42 @@ private ObjectAttributeChangeDetector() {}
     return changes.toString();
   }
 
-  private static void retrieveFields(List<Field> fields, Class<?> currentClass) {
-    while (currentClass.getSuperclass() != null) {
-      fields.addAll(Arrays.asList(currentClass.getDeclaredFields()));
-      currentClass = currentClass.getSuperclass();
+  private static JSONObject generateChangedAttribute(
+      Field field, Object oldValue, Object newValue) {
+    JSONObject changedAttribute = new JSONObject();
+    changedAttribute.put("fieldName", field.getName());
+    changedAttribute.put(
+        "oldValue", Optional.ofNullable(oldValue).map(JSONObject::wrap).orElse(""));
+    changedAttribute.put(
+        "newValue", Optional.ofNullable(newValue).map(JSONObject::wrap).orElse(""));
```

Comment:
@gitgoodjhe: The JSONObject#wrap is the "effective" change. Everything else is refactoring